set.mm

```diff
@@ -264980,12 +264978,11 @@ According to Wikipedia ("Integer", 25-May-2019,
                          -> ( Y ` ( S ` Q ) ) e. ( Base ` R ) ) $=
       ( cfv c1 wcel wceq eqid eqeltrd 3ad2ant1 fveq2 eleq1d syl5ibr neg1z wo wi
       cneg cpr crg cfn w3a cbs psgnran 3adant1 cur zrh1 ringidcl cmg co zrhmulg
-      elpri cz mpan2 cgrp ringgrp a1i mulgcl syl3anc jaoi syl mpcom ) BDJZKKUCZ
-      UDLZCUELZEUFLZBALZUGZVHFJZCUHJZLZVLVMVJVKABDEGHUIUJVJVHKMZVHVIMZUAVNVQUBZ
-      VHKVIUQVRVTVSVNVQVRKFJZVPLZVKVLWBVMVKWACUKJZVPCWCFIWCNZULVPCWCVPNZWDUMZOP
-      VRVOWAVPVHKFQRSVNVQVSVIFJZVPLZVKVLWHVMVKWGVIWCCUNJZUOZVPVKVIURLZWGWJMTCWI
-      WCFVIIWINZWDUPUSVKCUTLWKWCVPLWJVPLCVAWKVKTVBWFVPWICVIWCWEWLVCVDOPVSVOWGVP
-      VHVIFQRSVEVFVG $.
+      elpri cz mpan2 ringgrp a1i mulgcld jaoi syl mpcom ) BDJZKKUCZUDLZCUELZEUF
+      LZBALZUGZVFFJZCUHJZLZVJVKVHVIABDEGHUIUJVHVFKMZVFVGMZUAVLVOUBZVFKVGUQVPVRV
+      QVLVOVPKFJZVNLZVIVJVTVKVIVSCUKJZVNCWAFIWANZULVNCWAVNNZWBUMZOPVPVMVSVNVFKF
+      QRSVLVOVQVGFJZVNLZVIVJWFVKVIWEVGWACUNJZUOZVNVIVGURLZWEWHMTCWGWAFVGIWGNZWB
+      UPUSVIVNWGCVGWAWCWJCUTWIVITVAWDVBOPVQVMWEVNVFVGFQRSVCVDVE $.
 
```

Comment:
The theorem under focus here is zrhpsgnelbas.  Its proof length shrinks from 31 down to 30 lines.  Interestingly, the Syntax Definition Hints in the new proof do not list cgrp any more.  Why? Both proofs contain a line (𝑅 ∈ Ring → 𝑅 ∈ Grp) , so my expectation is that it is listed in the new proof, too.  The used Axioms and Definitions are the same.  The MM verifier accepts both proofs.

Reply:
If you look at both proofs with all steps (`MM> sh pr zrhpsgnelbas/ALL`), you see that the old proof needs to prove that `Grp` is a class in order to prove that `R e. Grp` is a formula. It is needed when used as `syl3anc.1`. In the new proof, that formula appears as `mulgcld.3`, where it is not necessary to prove that `R e. Grp` is a formula.